pkg/kp/healthmanager.go

```diff
@@ -269,7 +300,14 @@ func processHealthUpdater(
 			write = nil
 			if result.OK {
 				remoteHealth = result.Health
+				if result.Throttle && throttle == nil {
```

Comment:
My mistake, I do see there is a token bucket implementation - didn't fully grok that tokens were being added to the bucket, not removed, over time.